extensions/amp-smartlinks/0.1/linkmate.js

```diff
@@ -165,11 +163,29 @@ export class Linkmate {
    */
   getEditInfo_() {
     return dict({
-      'name': this.rootNode_.title || null,
-      'url': this.ampDoc_.getUrl(),
+      'name': this.getEditName_(),
+      'url': this.getLocationHref_(),
     });
   }
 
+  /**
+   * Retrieve edit name.
+   * @return {string}
+   * @private
+   */
+  getEditName_() {
+    return this.win_.document.getElementsByTagName('title')[0].text || null;
```

Comment:
Is it ok for you if the document doesn't have a title at all? In that case this line will fail. E.g. it will be the same as `emptyArray[0].title`.

Reply:
@dvoytenko I see what you mean. Just added a check to account for this!